matchzoo/preprocessor/embedding.py

```diff
@@ -0,0 +1,109 @@
+           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
+             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ],
+           [ 0.1       ,  0.2       ,  0.3       ,  0.4       ,  0.5       ,
+             0.6       ,  0.7       ,  0.8       ,  0.9       ,  1.        ]])
+        >>> embed_mat[0] # doctest: +SKIP
+        array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
+
+    """
+
+    def __init__(self, embed_file: str):
+        """
+        Class initialization.
+
+        :param embed_file: the embeding input file path.
+        """
+        self.embed_file = embed_file
+        self.embed_dim = 0
+        self.index_state = {}
+        self.embed_mat = None
+
+    def process(self, term_index: dict):
+        """Build a :np.array:`embed_mat` and a :dict:`index_state`."""
+        n_word_in_dict = max(term_index.values()) + 1
+        if n_word_in_dict != len(term_index):
+            logger.warning("Some words are not shown in term_index({}). Total "
+                           "number of words are {}.".format(len(term_index),
+                                                            n_word_in_dict))
+
+        # The number of word share with `term_index` and `embed_file`.
+        n_word_share = 0
+
+        with open(self.embed_file, 'rb') as embed_file_ptr:
+            # Read the header of the embed file
+            n_word_in_embed, self.embed_dim = \
+                map(int, embed_file_ptr.readline().rstrip().split(b" "))
+
+            self.embed_mat = np.random.uniform(-0.25, 0.25, (n_word_in_dict,
+                                                             self.embed_dim))
+
+            # Go through all embedding file
+            for line in tqdm(embed_file_ptr):
+                # Explicitly splitting on " " is important, so we don't
+                # get rid of Unicode non-breaking spaces in the vectors.
+                entries = line.rstrip().split(b" ")
+
+                word, vec = entries[0], entries[1:]
+
+                if self.embed_dim != len(vec):
+                    raise RuntimeError(
+                        "Vector for token {} has {} dimensions, but "
+                        "previously read vectors have {} dimensions. "
+                        "All vectors must have the same number of "
+                        "dimensions.".format(word, len(vec), self.embed_dim))
+
+                try:
+                    if isinstance(word, bytes):
+                        word = word.decode('utf-8')
+                except Exception as e:
+                    logger.warning(
+                        "Skipping non-UTF8 token {}".format(repr(word)))
+                    continue
+
+                if word in term_index:
+                    index = term_index[word]
+                    self.embed_mat[index] = \
+                        np.array([float(x) for x in vec])
+                    self.index_state[index] = 1
+                    n_word_share += 1
```

Comment:
I'd refactor it a little bit:
```python
index = term_index.get(word, None)
if index:
    self.embed_mat[index] = np.array(vec).astype(float) # use broadcasting instead of list comprehension.
    ...
    num_shared_word += 1
```